syntax = "proto3";

package blok.workflow.v1;

message NodeRequest {
    string Name = 1;
    string Message = 2;
    string Encoding = 3;
    string Type = 4;
}

message NodeResponse {
    string Message = 1;
    string Encoding = 2;
    string Type = 3;
}

enum MessageEncoding {
    BASE64 = 0;
    STRING = 1;
}

enum MessageType {
    TEXT = 0;
    JSON = 1;
    XML = 2;
    HTML = 3;
    BINARY = 4;
}

service NodeService {
    rpc ExecuteNode (NodeRequest) returns (NodeResponse) {}
}